src/main/java/net/minecraftforge/common/config/Configuration.java

```diff
@@ -80,9 +89,16 @@
 
     public Configuration(){}
 
+    @Deprecated
+    public Configuration(CharSource staticSource, CharSink staticSink){
```

Comment:
Apparently we fundamentally disagree how this should be done (I vote for abstraction instead of more special casing like you have done now, there shouldn't even be any `File` field in the class except for the getFile method.
But I've made my point.